Ocamldoc, latex backend: labels for exception

Octachron opened the pull request:
This brief PR restores the label for exceptions in the latex backend of ocamldoc (which fixes a handful of missing reference warnings in the latex version of the manual).

Octachron committed:
ocamldoc latex, labels for exceptions

This commit fixes the lack of label for exception definition in the latex
backend of ocamldoc.

gasche commented:
The label was removed by dcceb46358557d77079828c8730cb8168095330c. Could you add an exception declaration to the regression testsuite ?

Octachron commented:
This is clearly needed indeed. I have added an exception definition and some references to the extensible variants test for the latex backend, and fixed the reference file for the inlined record test .

gasche commented:
You need a Changes entry, as the label regression is present in released OCaml versions I think.

Octachron committed:
GPR#1457 Changes entry

Octachron commented:
I added the change  entry in the 4.06 section for now. Please tell me if I should move it to trunk.

gasche commented:
Cherry-picked in 4.06 ( edb532bbcbe76904d38a6fe461483c04a9825ad ).
